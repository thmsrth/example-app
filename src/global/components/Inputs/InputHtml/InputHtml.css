.InputHtml__input {
  border: 1px solid var(--border-color);
  padding: var(--section-padding);
  max-height: 300px;
}
