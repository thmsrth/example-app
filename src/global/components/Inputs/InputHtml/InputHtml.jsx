import React from 'react';
import PropTypes from 'prop-types';
import HtmlEditor from '../../Html/HtmlEditor';

import { bemCls } from '../../../utils/ClassNameHelpers';

import './InputHtml.css';

const InputHtml = ({
  className, name, label, value, onChange, onBlur, validationKeys, disabled,
}) => {
  const {
    required,
    validationAlert,
    warningAlert,
    onValidate,
  } = validationKeys || InputHtml.defaultProps.validationKeys;

  return (
    <div className={`InputHtml ${className} ${
      bemCls(`InputHtml ${className}`, `__${name}`)} ${
      bemCls(`InputHtml ${className}`, '__required', required)} ${
      bemCls(`InputHtml ${className}`, '--alert', validationAlert)}`}
    >
      <label
        key={name}
        className={`${bemCls(`InputHtml ${className}`, '__label')} ${
          bemCls(`InputHtml ${className}`, `__${name}__label`)} ${
          bemCls(`InputHtml ${className}`, '__label--disabled', disabled)}`}
      >
        {`${label}${required ? '*' : ''}`}
        <div className={`${bemCls(`InputHtml ${className}`, '__alertMsg')} ${
          bemCls(`InputHtml ${className}`, `__${name}__alertMsg`)}`}
        >
          {validationAlert}
        </div>
        <HtmlEditor
          className={`${bemCls(`InputHtml ${className}`, '__input')} ${
            bemCls(`InputHtml ${className}`, `__${name}__input`)} ${
            bemCls(`InputHtml ${className}`, '__input--disabled', disabled)}`}
          name={name}
          value={value}
          required={required}
          disabled={disabled}
          onChange={onChange}
          onBlur={onBlur}
          onValidate={onValidate}
        />
      </label>
      <div className={`${bemCls(`InputHtml ${className}`, '__warningMsg')} ${
        bemCls(`InputHtml ${className}`, `__${name}__warningMsg`)}`}
      >
        {warningAlert}
      </div>
    </div>
  );
};

export default InputHtml;

InputHtml.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string,
  className: PropTypes.string,
  value: PropTypes.string,
  validationKeys: PropTypes.shape({
    required: PropTypes.bool,
    validationAlert: PropTypes.string,
    warningAlert: PropTypes.string,
    onValidate: PropTypes.func,
  }),
  disabled: PropTypes.bool,
  onChange: PropTypes.func,
  onBlur: PropTypes.func,
};

InputHtml.defaultProps = {
  label: '',
  className: '',
  value: '',
  validationKeys: {
    required: false,
    validationAlert: '',
    warningAlert: '',
    onValidate: () => {},
  },
  disabled: false,
  onChange: () => {},
  onBlur: () => {},
};
