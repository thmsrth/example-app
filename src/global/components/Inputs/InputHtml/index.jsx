import React, { Component } from 'react';
import PropTypes from 'prop-types';

import InputHtml from './InputHtml';

class InputHtmlWrapper extends Component {
  constructor() {
    super();
    this.handleChange = this.handleChange.bind(this);
    this.handleBlur = this.handleBlur.bind(this);
  }

  handleChange(html) {
    const { name, onChange } = this.props;
    onChange(name, html);
  }

  handleBlur() {
    const {
      name, validationKeys, onBlur,
    } = this.props;
    const options = { type: 'html' };

    onBlur(name, validationKeys, options);
  }

  render() {
    const { value, hidden, ...otherProps } = this.props;

    if (hidden) return null;

    return (
      <InputHtml
        {...otherProps}
        value={value}
        onChange={this.handleChange}
        onBlur={this.handleBlur}
      />
    );
  }
}

export default InputHtmlWrapper;

InputHtmlWrapper.propTypes = {
  value: PropTypes.string,
  name: PropTypes.string,
  validationKeys: PropTypes.shape({
    required: PropTypes.bool,
    pattern: PropTypes.instanceOf(RegExp),
    validationAlert: PropTypes.string,
    warningAlert: PropTypes.string,
    onValidate: PropTypes.func,
    onWarning: PropTypes.func,
  }),
  onChange: PropTypes.func,
  onBlur: PropTypes.func,
  hidden: PropTypes.bool,
};

InputHtmlWrapper.defaultProps = {
  value: '',
  name: '',
  validationKeys: null,
  onChange: () => {},
  onBlur: () => {},
  hidden: false,
};
