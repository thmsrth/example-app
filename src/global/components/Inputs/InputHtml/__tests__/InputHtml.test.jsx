import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import InputHtml from '../InputHtml';

const props = {
  className: 'MyInputText',
  name: 'mytext',
  label: 'My Input Label',
  value: 'My Text',
  validationKeys: {
    required: false,
    validationAlert: '',
    warningAlert: '',
  },
  onChange: () => {},
};

describe('InputHtml', () => {
  it('should render the component correctly with type text', () => {
    const tree = shallow(<InputHtml {...props} />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should render the component as a required input', () => {
    const tree = shallow(<InputHtml
      {...props}
      validationKeys={{ required: true }}
    />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should render the component with a validation alert message', () => {
    const tree = shallow(<InputHtml
      {...props}
      validationKeys={{ validationAlert: 'My validation alert message' }}
    />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should render the component with a warning message', () => {
    const tree = shallow(<InputHtml
      {...props}
      validationKeys={{ warningAlert: 'My warning message' }}
    />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
