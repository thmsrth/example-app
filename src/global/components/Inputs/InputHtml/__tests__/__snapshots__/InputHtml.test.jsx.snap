// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputHtml should render the component as a required input 1`] = `
<div
  className="InputHtml MyInputText InputHtml__mytext MyInputText__mytext InputHtml__required MyInputText__required InputHtml--alert MyInputText--alert"
>
  <label
    className="InputHtml__label MyInputText__label InputHtml__mytext__label MyInputText__mytext__label "
    key="mytext"
  >
    My Input Label*
    <div
      className="InputHtml__alertMsg MyInputText__alertMsg InputHtml__mytext__alertMsg MyInputText__mytext__alertMsg"
    />
    <HtmlEditorView
      className="InputHtml__input MyInputText__input InputHtml__mytext__input MyInputText__mytext__input "
      disabled={false}
      name="mytext"
      onBlur={[Function]}
      onChange={[Function]}
      onValidate={[Function]}
      readOnly={false}
      required={true}
      value="My Text"
    />
  </label>
  <div
    className="InputHtml__warningMsg MyInputText__warningMsg InputHtml__mytext__warningMsg MyInputText__mytext__warningMsg"
  />
</div>
`;

exports[`InputHtml should render the component correctly with type text 1`] = `
<div
  className="InputHtml MyInputText InputHtml__mytext MyInputText__mytext  "
>
  <label
    className="InputHtml__label MyInputText__label InputHtml__mytext__label MyInputText__mytext__label "
    key="mytext"
  >
    My Input Label
    <div
      className="InputHtml__alertMsg MyInputText__alertMsg InputHtml__mytext__alertMsg MyInputText__mytext__alertMsg"
    />
    <HtmlEditorView
      className="InputHtml__input MyInputText__input InputHtml__mytext__input MyInputText__mytext__input "
      disabled={false}
      name="mytext"
      onBlur={[Function]}
      onChange={[Function]}
      onValidate={[Function]}
      readOnly={false}
      required={false}
      value="My Text"
    />
  </label>
  <div
    className="InputHtml__warningMsg MyInputText__warningMsg InputHtml__mytext__warningMsg MyInputText__mytext__warningMsg"
  />
</div>
`;

exports[`InputHtml should render the component with a validation alert message 1`] = `
<div
  className="InputHtml MyInputText InputHtml__mytext MyInputText__mytext InputHtml__required MyInputText__required InputHtml--alert MyInputText--alert"
>
  <label
    className="InputHtml__label MyInputText__label InputHtml__mytext__label MyInputText__mytext__label "
    key="mytext"
  >
    My Input Label
    <div
      className="InputHtml__alertMsg MyInputText__alertMsg InputHtml__mytext__alertMsg MyInputText__mytext__alertMsg"
    >
      My validation alert message
    </div>
    <HtmlEditorView
      className="InputHtml__input MyInputText__input InputHtml__mytext__input MyInputText__mytext__input "
      disabled={false}
      name="mytext"
      onBlur={[Function]}
      onChange={[Function]}
      onValidate={[Function]}
      readOnly={false}
      required={false}
      value="My Text"
    />
  </label>
  <div
    className="InputHtml__warningMsg MyInputText__warningMsg InputHtml__mytext__warningMsg MyInputText__mytext__warningMsg"
  />
</div>
`;

exports[`InputHtml should render the component with a warning message 1`] = `
<div
  className="InputHtml MyInputText InputHtml__mytext MyInputText__mytext InputHtml__required MyInputText__required InputHtml--alert MyInputText--alert"
>
  <label
    className="InputHtml__label MyInputText__label InputHtml__mytext__label MyInputText__mytext__label "
    key="mytext"
  >
    My Input Label
    <div
      className="InputHtml__alertMsg MyInputText__alertMsg InputHtml__mytext__alertMsg MyInputText__mytext__alertMsg"
    />
    <HtmlEditorView
      className="InputHtml__input MyInputText__input InputHtml__mytext__input MyInputText__mytext__input "
      disabled={false}
      name="mytext"
      onBlur={[Function]}
      onChange={[Function]}
      onValidate={[Function]}
      readOnly={false}
      required={false}
      value="My Text"
    />
  </label>
  <div
    className="InputHtml__warningMsg MyInputText__warningMsg InputHtml__mytext__warningMsg MyInputText__mytext__warningMsg"
  >
    My warning message
  </div>
</div>
`;
